docs/sphinx/examples/qec/python/generate_parity_check_matrix.py

```diff
@@ -0,0 +1,140 @@
+import cudaq
+import cudaq_qec as qec
+import numpy as np
+
+# Set the number of shots and rounds
+nShots = 30
+nRounds = 5
+distance = 4
+cudaq.set_target("stim")
+
+
+def construct_measurement_error_syndrome(distance, nRounds):
+    num_stabilizers = distance - 1
+    num_mea_q = num_stabilizers * nRounds
+
+    syndrome_rows = []
+
+    # In this scheme, need two rounds for each measurement syndrome
+    for i in range(nRounds - 1):
+        for j in range(num_stabilizers):
+            syndrome = np.zeros((num_mea_q,), dtype=np.int8)
+
+            # The error on ancilla (j) in round (i) affects stabilizer checks at two positions:
+            # First occurrence in round i
+            pos1 = i * num_stabilizers + j
+            # Second occurrence in round i+1
+            pos2 = (i + 1) * num_stabilizers + j
+
+            # Mark the syndrome
+            syndrome[pos1] = 1
+            syndrome[pos2] = 1
+
+            syndrome_rows.append(syndrome)
+
+    return np.array(syndrome_rows).T
+
+
+def get_code_and_pcm(distance, nRounds):
+    if nRounds < 2:
+        raise ValueError("nRounds must be greater than or equal to 2")
+    if distance < 3:
+        raise ValueError("distance must be greater than or equal to 3")
+
+    # Create distance-3 repetition code
+    code = qec.get_code("repetition", distance=distance)
+
+    # Get the parity check matrix for a single round
+    Hz = code.get_parity_z()
+    H = np.array(Hz)
+
+    # Extends H to nRounds
+    rows, cols = H.shape
+    H_nrounds = np.zeros((rows * nRounds, cols * nRounds), dtype=int)
+    for i in range(nRounds):
+        H_nrounds[i * rows:(i + 1) * rows, i * cols:(i + 1) * cols] = H
+    print("H_nrounds\n", H_nrounds)
+
+    # Append columns for measurement errors to H
+    Mz = construct_measurement_error_syndrome(distance, nRounds)
+    print("Mz\n", Mz)
+    assert H_nrounds.shape[0] == Mz.shape[
+        0], "Dimensions of H_nrounds and Mz do not match"
+    H_pcm = np.concatenate((H_nrounds, Mz), axis=1)
+    print(f"H_pcm:\n{H_pcm}")
+
+    return code, H_pcm, Mz
+
+
+code, H_pcm, Mz = get_code_and_pcm(distance, nRounds)
+
+# Define the error probability
+p = 0.01
+noise = cudaq.NoiseModel()
+noise.add_all_qubit_channel("x", qec.TwoQubitDepolarization(p), 1)
```

Comment:
If it still works, it might be nice to show how to add measurement noise in this example.
```suggestion
noise.add_all_qubit_channel("x", qec.TwoQubitDepolarization(p), 1)
noise.add_all_qubit_channel("mz", cudaq.BitFlipChannel(p))
```